src/d3a/models/area/redis_external_market_connection.py

```diff
@@ -40,64 +48,104 @@ def _market_stats_channel(self):
     def _grid_fees_channel(self):
         return f"{self.channel_prefix}/grid_fees"
 
-    def sub_to_area_event(self):
-        self.redis_db = self.area.config.external_redis_communicator
-        self.redis_db.sub_to_multiple_channels({
+    @staticmethod
+    def _get_transaction_id(payload):
+        data = json.loads(payload["data"])
+        if "transaction_id" in data and data["transaction_id"] is not None:
+            return data["transaction_id"]
+        else:
+            raise ValueError("transaction_id not in payload or None")
+
+    def _register(self, payload):
+        self._connected = register_area(self.redis_com, self.channel_prefix, self.connected,
+                                        self._get_transaction_id(payload),
+                                        area_uuid=self.area.uuid)
+
+    def _unregister(self, payload):
+        self._connected = unregister_area(self.redis_com, self.channel_prefix, self.connected,
+                                          self._get_transaction_id(payload))
+
+    def sub_to_external_channels(self):
+        self.redis_com = self.area.config.external_redis_communicator
+        sub_channel_dict = {
             f"{self.channel_prefix}/market_stats": self.market_stats_callback,
             f"{self.channel_prefix}/dso_market_stats": self.dso_market_stats_callback,
-            f"{self.channel_prefix}/grid_fees": self.set_grid_fees_callback
-        })
+            f"{self.channel_prefix}/grid_fees": self.set_grid_fees_callback,
+            f"{self.channel_prefix}/register_participant": self._register,
+            f"{self.channel_prefix}/unregister_participant": self._unregister}
+        if self.area.config.external_redis_communicator.is_enabled:
+            self.aggregator = self.area.config.external_redis_communicator.aggregator
+        self.redis_com.sub_to_multiple_channels(sub_channel_dict)
 
     def market_stats_callback(self, payload):
         market_stats_response_channel = f"{self.channel_prefix}/response/market_stats"
-        payload_data = json.loads(payload["data"])
+        payload_data = json.loads(payload["data"]) \
+            if isinstance(payload["data"], str) else payload["data"]
```

Comment:
I am curious why this is needed, would be great if you took some time to explain this. I also stumbled upon a similar change in a d3a-web PR, could it be that these 2 are related ?

Reply:
IIRC, this is needed for the batch_commands. Here the individual commands are already parsed dicts, but if this method is called directly via the endpoint, the payload["data"] is a string.

Reply:
Hmm, I assume you mean this method `trigger_aggregator_commands` when you mention `but if this method is called directly via the endpoint,` . How about doing a json.loads from the trigger_aggregator_commands method, so that the `market_stats_callback` accepts only dicts ?

Reply:
I will actually have a look again why this is different for the devices and markets. Will keep you posted

Reply:
Thanks for checking it out!

Reply:
I found out that it is actually the other way around. for the individual commands payload["data"] is a string, but for the batch commands, it is already deserialised to a dict. In the devices, we have different functions for the aggregator and for the direct commands. Here, i propose to keep the if/then implementation and not implement an extra function. Do you agree?

Reply:
Agreed, in this case there is no need to change the implementation